forge/ee/lib/billing/trialTask.js

```diff
@@ -0,0 +1,133 @@
+const { Op } = require('sequelize')
+const { KEY_BILLING_STATE } = require('../../../db/models/ProjectSettings')
+
+module.exports.init = function (app) {
+    async function trialTask (app) {
+        // Find teams that have expired since we last ran (trialEndsAt != null && < now)
+        const expiredSubscriptions = await app.db.models.Subscription.findAll({
+            where: { trialEndsAt: { [Op.lt]: Date.now() } },
+            include: [app.db.models.Team]
+        })
+
+        for (const subscription of expiredSubscriptions) {
+            if (subscription.isActive()) {
+                // The subscription has been setup on Stripe.
+                // Add all trial projects to billing.
+                const trialProjects = await addTrialProjectsToBilling(subscription.Team)
+                let trialProjectName
+                if (trialProjects.length > 0) {
+                    trialProjectName = trialProjects[0].name
+                }
+                await sendTrialEmail(subscription.Team, 'TrialTeamEnded', {
+                    trialProjectName
+                })
+            } else {
+                // Stripe not configured - suspend the lot
+                await suspendAllProjects(subscription.Team)
+                await sendTrialEmail(subscription.Team, 'TrialTeamSuspended', {
+                    teamSettingsURL: `${app.config.base_url}/team/${subscription.Team.slug}/billing`
+                })
+            }
+
+            // We have dealt with this team
+            subscription.trialEndsAt = null
+            subscription.trialStatus = app.db.models.Subscription.TRIAL_STATUS.ENDED
+            await subscription.save()
+        }
+
+        // Email sending intervals
+        // - 8 days => Subscription.status = TRIAL_STATUS.WEEK_EMAIL_SENT
+        // - 2 days => Subscription.status = TRIAL_STATUS.DAY_EMAIL_SENT
+
+        const pendingEmailSubscriptions = await app.db.models.Subscription.findAll({
+            where: {
+                [Op.or]: [
+                    {
+                        // Expires in 8 days, need to send Week Reminder email
+                        trialStatus: app.db.models.Subscription.TRIAL_STATUS.CREATED,
+                        trialEndsAt: { [Op.lt]: Date.now() + (8 * 86400000) }
```

Comment:
I'd swap `86400000` for a var like `const ONE_DAY = 86400000`

Reply:
done